src/realm/array_integer.hpp

```diff
@@ -70,6 +70,14 @@ class ArrayInteger : public Array, public ArrayPayload {
     }
     template <class cond>
     bool find(value_type value, size_t start, size_t end, QueryStateBase* state) const;
+
+    template <class T>
+    static ref_type typed_write(ref_type ref, T& out, Allocator& alloc)
+    {
+        Array arr(alloc);
```

Comment:
This is a new path in the code, we are constructing a new integer array, ending up calling `init_from_mem`, which will eventually parse the header in order to find out if the array is compressed. I don't think it is going to slow down the commit too much (not more than what we already are).. But also this is not related to this PR. 
I'll take care of measuring all again.

Reply:
Not sure there is a difference. You also called init_from_ref in your original implementation. I am just calling it in a different place in the code.

Reply:
yep, I am not sure if it is actually making a difference.